cluster_pack/packaging.py

```diff
@@ -408,6 +420,33 @@ def get_editable_requirements(
     return editable_requirements
 
 
+def get_pyenv_usage_from_archive(path_to_archive: str) -> PythonEnvDescription:
+
+    archive_filename = os.path.basename(path_to_archive)
+
+    if archive_filename.endswith('.pex.zip'):
+        return PythonEnvDescription(
+            path_to_archive,
+            f"{LARGE_PEX_CMD}",
+            UNPACKED_ENV_NAME,
+            True)
+    elif archive_filename.endswith('.pex'):
+        return PythonEnvDescription(
+            path_to_archive,
+            f"./{archive_filename}",
+            archive_filename,
+            False)
+    elif archive_filename.endswith(".zip") or archive_filename.endswith(".tar.gz"):

```

Comment:
In fact this condition only works if we have already checked that the file was not ending with '.pex.zip' first, so I'd prefer to have explicit tests on all possible extensions instead of hiding a part of the resolution in a function.

I've only found 2 usages of this condition (here and in `detect_packer_from_file`), which made me realize I also had to modify `detect_packer_from_file` to handle '.pex.zip' also

Reply:
Fair, let's leave it as is